pkg/cli/admin/prune/deployments/prune.go

```diff
@@ -46,6 +47,10 @@ type PrunerOptions struct {
 	DeploymentConfigs []*appsv1.DeploymentConfig
 	// Deployments is the entire list of deployments across all namespaces in the cluster.
 	Deployments []*corev1.ReplicationController
+	// V1Deployments is the entire list of v1 Deployments across all namespaces in the cluster.
+	V1Deployments []*k8sappsv1.Deployment
```

Comment:
Deployments (+RS) and DeploymentConfigs (+RC) are two distinct sets, so I'd suggest creating entirely separate Pruner just for the former.

Reply:
Going through pruner and especially resolver I see they are very tightly coupled with openshift types, you could try to generlize the code to work with both but it'll be similar effort to writing entirely separate just for k8s types. The latter will be easier to understand, maintain and test. In the `PruneDploymentOptions`'s `Run` method you'd just call one and then the other, let's start with this simpler approach and we can iterate on it as we go.